### ISSUE TITLE
Epic: Support dictionary/data lookup in MacVim

### ISSUE THREAD

**Issue by ychin** [macvim-dev/macvim#1311]
It is useful to support the ability to look up a piece of selected / highlighted text in MacVim and show a OS popup that gives you more information. #1191 was the original issue, but this work has ballooned as this becomes more ambitious and need to span multiple commits. Using this epic to track the work and elicit ideas / brainstorm.

**Comment by eirnym** [comment]
I wonder if there will be an option to disable this feature

**Comment by ychin** [comment]
Why would you need to disable it? This only triggers per user interaction. If you don’t like the built in dictionary you can just not trigger the lookup which is quite hard to accidentally do.

**Comment by eirnym** [comment]
Yep, I don't want to trigger is dictionary lookup. Dictionary for Polish, primary language I use beside English during writing a code is so poor and I there's no way to install a custom one, so I don't use this feature in macOS at all as it's completely useless.

Additionally, I like MacVim to be a editor with lightweight OS integration when I don't need it. Also I like Vim itself for an ability to customise it. 

As a possible solution I propose make operations noop. I agree, that custom integration is a good thing, but not the macOS dictionary.

**Comment by ychin** [comment]
I still don’t understand under what circumstance would this be an issue for you? Can you be more specific? I’m not asking why you don’t like the built in dictionary. Im asking why we would need an option to turn this off.

Only known ways I know to trigger this:
* Ctrl-Cmd-D: this is super specific and I doubt something people accidentally trigger too much. It’s OS-wide so if you dislike the dictionary so much I would imagine you would have avoided this keystroke a long time ago
* Trackpad: either force click or three finger taps, both of which are not easy to misfire *and* the OS already allows you to turn it off system-wide.

I just don’t understand why you would care if you never use this feature. Most macOS applications support this so this is a pretty basic OS integration. I think options make sense when there is a reason for them but not when the feature is completely opt in to begin with.
